UI/src/JS/script.js

```diff
@@ -0,0 +1,508 @@
+                let newParagraph = document.createElement('p');
+                newParagraph.className = 'alert-wrong-data';
+                newParagraph.textContent = "fields can't be empty!";
+                newParagraph.style.marginBottom = '1rem';
+                form.querySelector(".sign-in-button").after(newParagraph);
+            }
+        }
+    }
+    document.querySelector(".site-name-button").addEventListener("click", handleClickOnSiteName);
+    function handleClickOnSiteName(){
+        view.siteNameClicked();
+        view.showAll(skip, top, filterConfig);
+    }
+    document.querySelector(".offers-list").addEventListener("click", handleClickOnOffers);
+    function deleteOffer(button){
+        let id = button.closest('article').getAttribute('offer-id');
+        view.deleteAd(id);
+        view.showAll(skip, top, filterConfig);
+    }
+    function handleSubmitEditOffer(){
+        event.preventDefault();
+
+        let form = event.target;
+        let newOffer = {};
+        newOffer.label = form.elements["label-input"].value;
+        newOffer.description = form.elements["description-input"].value;
+        newOffer.link = form.elements["link-input"].value;
+        if(form.elements["date-input"].value !== ''){
+            newOffer.validUntil = new Date(form.elements["date-input"].value);
+        }
+        newOffer.discount = form.elements["discount-input"].value;
+        newOffer.photoLink = form.elements["photo-input"].value;
+        if(form.elements["hashtags-input"].value !== ''){
+            newOffer.hashTags = form.elements["hashtags-input"].value.split(' ');
+        }
+        view.editAd(editingId, newOffer);
+        handleClickOnSiteName();
+    }
+    function editOffer(button){
+        editingId = event.target.closest('article').getAttribute('offer-id');
+        view.addOfferClicked();
+        if(document.querySelector(".add-offer-main-page") != null){
+            document.forms[0].addEventListener('submit', handleSubmitEditOffer);
+        }
+    }
+    function handleSubmitAddComment(){
+        event.preventDefault();
+        let form = event.target;
+        let review = {}
+        if(form.elements.reviewInput.value != '' && form.elements.ratingInput.value != ''){
+            review.text = form.elements.reviewInput.value;
+            review.rating = parseInt(form.elements.ratingInput.value);
+            if(review.rating > 5 || review.rating < 1){
+                if(form.querySelector('.alert-wrong-data') != null){
+                    form.querySelector('.alert-wrong-data').remove();
+                }
+                let newParagraph = document.createElement('p');
+                newParagraph.className = 'alert-wrong-data';
+                newParagraph.textContent = "rating must be 1-5!";
+                newParagraph.style.marginBottom = '1rem';
+                form.querySelector(".add-review-button").after(newParagraph);
```

Comment:
I'd advice to at least move this block to a separate folder along with lines 468-472, just use newParagraph.textContent as a parameter of function.